new-components/targets/git-clone/pkg/git/git.go

```diff
@@ -0,0 +1,207 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/go-errors/errors"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/smithy-security/pkg/env"
+)
+
+const errInvalidConfigurationReasonEmpty = "cannot be empty"
+
+type (
+	// Conf wraps the component configuration.
+	Conf struct {
+		ClonePath  string
+		RepoURL    string
+		BranchName string
+
+		ConfAuth ConfAuth
+	}
+
+	// ConfAuth contains authentication configuration.
+	ConfAuth struct {
+		Username    string
+		AuthEnabled bool
```

Comment:
the default git auth is private key, not necessary now but let's add a ticket for this please

Reply:
Done! https://linear.app/smithy/issue/OCU-272/support-private-keys-for-new-git-clone-component